pages/reset.tsx

```diff
@@ -0,0 +1,247 @@
+import React, { useState } from "react";
+import { Flex, Box, Heading, Text, Center } from "@chakra-ui/react";
+import isEmail from "validator/lib/isEmail";
+import { ChevronLeftIcon } from "@chakra-ui/icons";
+import { Link } from "@chakra-ui/react";
+
+import { TextInput, PasswordInput, AuthButton } from "@components";
+
+const ResetPassword: React.FC = () => {
+    const [currStage, setCurrStage] = useState(0);
+    const [email, setEmail] = useState("");
+    const [emailInvalid, setEmailInvalid] = useState({
+        isInvalid: false,
+        reason: "",
+    });
+    const [passwordRequirements, setPasswordRequirements] = useState({
+        isInvalid: false,
+        reason: "",
+    });
+    const [passwordInvalid, setPasswordInvalid] = useState({
+        isInvalid: false,
+        reason: "",
+    });
+    const [newPassword, setNewPassword] = useState("");
+    const [confirmPassword, setConfirmPassword] = useState("");
+    const [canContinue, setCanContinue] = useState(false);
+
+    const validateEmail = () => {
+        if (isEmail(email)) {
+            //TODO: check that an account exists
+            setEmailInvalid({ isInvalid: false, reason: "" });
+            setCanContinue(true);
+        } else {
+            setEmailInvalid({
+                isInvalid: true,
+                reason: "Invalid Email",
+            });
+            setCanContinue(false);
+        }
+    };
+
+    const validatePassword = () => {
+        //TODO: update password validation requirements
+        if (newPassword.length > 7) {
+            setPasswordRequirements({ isInvalid: false, reason: "" });
+        } else {
+            setPasswordRequirements({
+                isInvalid: true,
+                reason: "Password does not meet requirements",
+            });
+            setCanContinue(false);
+        }
+    };
+
+    const incrementStage = () => {
+        // add logic to check if it's possible
```

Comment:
is this todo? otherwise can remove (lol this is also probably transferred from my old wip)

Reply:
I think we may want validation later about like has the page been completed before moving on- but I think we prevent that by disabling the button so we are fine?